### PR TITLE
14/play pause

### LINKED ISSUE #7: add grid

### LINKED ISSUE #11: Handle degrees Kelvin and Celsius units

### LINKED ISSUE #13: Label Thermometer with Earth, March, etc.

### LINKED ISSUE #14: Add Play and Pause buttons

### LINKED ISSUE #16: Add switch for macroscopic / microscopic view

### PR DESCRIPTION
**Pull request by pyphilia**
close #7 
close #16 
close #14 
close #13 
close #11

### CODE REVIEW

File: src/utils/utils.js [hunk 1]
```diff
@@ -30,3 +30,7 @@ export const findXPositionsOfPositiveIons = (
       leftIndent + positiveIonRadius + widthOfPositiveIonWithPadding * index,
   );
 };
+
```
Comment (juancarlosfarah) [inline src/utils/utils.js:33]: These conversations are wrong. Do you mean Celsius to Kelvin and vv?

File: src/components/lab/thermometer/Scale.js [hunk 2]
```diff
@@ -49,59 +198,67 @@ const Scale = ({
...
-  // select marks at most number of scale we can display
-  const maxNbScale = Math.floor(thermometerHeight / SCALE_HEIGHT);
-  const prop = Math.ceil(scaleArray.length / maxNbScale);
-  if (prop > 1) {
-    scaleArray = scaleArray.filter((_, i) => i % prop === 0);
-  }
+  const labelYOffset = offsetY - SCALE_LEGEND_PADDING_BOTTOM;
+
+  // build fahrenheit scales
+  const FahrenheightScaleComponents = buildFahrenheitScales({
```
Comment (juancarlosfarah) [inline src/components/lab/thermometer/Scale.js:207]: Typo in Fahrenheit. But I think you mean Kelvin (from the conversions below).

File: src/components/lab/Ground.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        fillLinearGradientStartPoint={{ x: 0, y: groundHeight }}
+        fillLinearGradientEndPoint={{ x: 0, y: 0 }}
+        fillLinearGradientColorStops={[
+          0,
+          GROUND_FILL_DARK_COLOR,
+          1,
+          GROUND_FILL_LIGHT_COLOR,
+        ]}
+      />
+      <Line x={0} y={0} points={points} tension={0.3} closed fill="#34852d" />
```
Comment (juancarlosfarah) [inline src/components/lab/Ground.js:58]: Should we factor out the tension?

**Review by juancarlosfarah — CHANGES_REQUESTED** [review]
Looks good! But I have a question on the Fº, Cº, and K temperature scales. Please have a look.

**Comment by pyphilia** [comment]
Indeed I called it fahrenheit but they were Kelvin... I'll rename everything correctly.

File: src/components/lab/Lattice.js [hunk 4]
```diff
@@ -27,22 +30,50 @@ class Lattice extends Component {
...
+        const oscillation = IONS_OSCILLATION_RADIUS * Math.sin(angle);
+
+        this.setState({
+          oscillation,
+          angle:
+            // transform from scale to float
+            // add an offset to allow oscillation even at minumum temperature
+            angle +
+            (((temperature + IONS_OSCILLATION_OFFSET - from) * 1.0) / to) *
+              IONS_OSCILLATION_SPEED_FACTOR,
```
Comment (hagoptaminian) [inline src/components/lab/Lattice.js:71]: I believe the `/ to` here has caused the problem that the charges are no longer oscillating (the resulting angle is too small?). Not sure why it was changed from previous commits, but this needs revision.

Also, why the `* 1.0`?

File: src/components/common/SideMenu.js [hunk 5]
```diff
@@ -121,23 +191,37 @@ class SideMenu extends React.Component {
...
+  isPaused: lab.isPaused,
+  isMicroscopic: lab.isMicroscopic,
+  showThermometerLabels: lab.showThermometerLabels,
 });
 
 const mapDispatchToProps = {
   dispatchToggleSideMenu: toggleSideMenu,
   dispatchToggleElectrons: toggleElectrons,
   dispatchToggleSpectrumBar: toggleSpectrumBar,
+  dispatchSetIsPause: setIsPaused,
```
Comment (hagoptaminian) [inline src/components/common/SideMenu.js:222]: `dispatchSetIsPaused`

File: src/components/lab/Lab.js [hunk 6]
```diff
@@ -43,6 +45,8 @@ class Lab extends Component {
       stageHeight: PropTypes.number.isRequired,
     }).isRequired,
     spectrumBar: PropTypes.bool.isRequired,
+    isMicroscopic: PropTypes.string.isRequired,
```
Comment (hagoptaminian) [inline src/components/lab/Lab.js:48]: `PropTypes.bool.isRequired`

File: src/components/lab/Lattice.js [hunk 7]
```diff
@@ -18,7 +19,9 @@ class Lattice extends Component {
     temperature: PropTypes.number.isRequired,
     scales: PropTypes.shape({
       from: PropTypes.number.isRequired,
+      to: PropTypes.string.isRequired,
```
Comment (hagoptaminian) [inline src/components/lab/Lattice.js:22]: `PropTypes.number.isRequired`

File: src/utils/utils.js [hunk 8]
```diff
@@ -30,3 +30,7 @@ export const findXPositionsOfPositiveIons = (
...
+export const kelvinToCelsius = (f) => f - 273.15;
```
Comment (hagoptaminian) [inline src/utils/utils.js:34]: now `(k) => k - 273.15`

File: src/reducers/lab.js [hunk 9]
```diff
@@ -1,11 +1,19 @@
...
+  SET_IS_MICROSCOPIC,
+  SET_SHOW_THERMOMETER_LABELS,
+} from '../types';
 
 const INITIAL_STATE = {
-  temperature: 300,
+  temperature: 400,
   scales: {
     to: 800,
     from: 200,
```
Comment (hagoptaminian) [inline src/reducers/lab.js:12]: factor out to constants?

INITIAL_TEMPERATURE, etc?

**Review by hagoptaminian — CHANGES_REQUESTED** [review]
it's looking really really nice. love this app! a few more small comments on files not in here:

- in `PositiveIon.js`, remove `thermometer` from `propTypes` (it's no longer being passed as a prop)
- do you think you will need to add some kind of indication that one grid unit is 100nm? (to help with the measurement)
- IMO it would be nice to make the `SpectrumBar.js` draggable (with a grab icon), but it's your call

File: src/components/lab/Lattice.js (hunk 4 above)
Comment (pyphilia) [reply src/components/lab/Lattice.js:71]: `*1.0` was to transform the initial temperature to a float. Which will result in a float after the division of the integer `to`. But I changed it to include the float part in the constant.

I revise a bit the computation, but the idea was to use `/to` to normalize the result, then have a more "neutral" speed factor.

**Review by hagoptaminian — APPROVED** [review]
👍 👍

**Pull request merged**